src/arrayMethodSort.js

```diff
@@ -5,7 +5,48 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    let compare = compareFunction;
+
+    // compare function default - comparing as strings
+    if (compare === undefined
+      || typeof compare !== 'function') {
+      compare = (a, b) => {
+        const _a = String(a);
+        const _b = String(b);
+
+        // not sure why localCompare would not work for cyryllic letters
+        if (_a > _b) {
+          return 1;
+        }
+
+        if (_b > _a) {
+          return -1;
+        }
+
+        return 0;
+      };
+    }
```

Comment:
You can move these lines of code to the new function; that way, the code will be clearer.